### PR TITLE
All i wanted was to add brass windoors but i ended tweaking windoor code.

### PR DESCRIPTION
**Pull request by jknpj**
* You don't need to disassemble windoors completely to move them anymore.
* Order doesn't really matter anymore when dis/assembling windoors. They need to be anchored to get their electronics/wires messed with though
* Emagging now leaves a fingerprint.
* Fixed a few unreport issues
* The overall code is still bad
* Added brass windoors, and its recipe
* Stray clockwork door/effect too because i gave up adding them in this PR.

:cl:
 * tweak: Windoors can now be unwrenched without needing to be completely disassembled.
 * rscadd: Adds brass windoors and brass windoor recipe.

### CODE REVIEW

File: code/game/machinery/doors/windowdoor.dm [hunk 1]
```diff
@@ -232,35 +224,34 @@
 
 /obj/machinery/door/window/attackby(obj/item/weapon/I as obj, mob/living/user as mob)
 	// Make emagged/open doors able to be deconstructed
-	if (!density && operating != 1 && iscrowbar(I))
-		user.visible_message("[user] removes the electronics from the windoor assembly.", "You start to remove the electronics from the windoor assembly.")
+	if(!density && operating != 1 && iscrowbar(I))
+		user.visible_message("[user] removes [electronics] from [src].", "You start to remove [electronics] from [src].")
```
Comment (MadmanMartian) [inline code/game/machinery/doors/windowdoor.dm]: use dem \the's

File: code/game/machinery/doors/windowdoor.dm [hunk 2]
```diff
@@ -232,35 +224,34 @@
...
 		playsound(src, 'sound/items/Crowbar.ogg', 100, 1)
-		if (do_after(user, src, 40) && src && !density && operating != 1)
-			to_chat(user, "<span class='notice'>You removed the windoor electronics!</span>")
+		if(do_after(user, src, 40) && src && !density && operating != 1)
+			to_chat(user, "<span class='notice'>You removed [electronics]!</span>")
```
Comment (MadmanMartian) [inline code/game/machinery/doors/windowdoor.dm]: use dem \the's

File: code/game/objects/structures/windoor_assembly.dm [hunk 3]
```diff
@@ -66,253 +67,214 @@ obj/structure/windoor_assembly/Destroy()
...
+		user.visible_message("[user] is prying [src] into the frame.", "You start prying [src] into the frame.")
+
+		if(do_after(user, src, 40))
+			if(gcDestroyed)
+				return
+			var/spawn_type = secure ? secure_type : windoor_type
+			var/obj/machinery/door/window/windoor = new spawn_type(loc)
+			to_chat(user, "<span class='notice'>You finish the [windoor.name]!</span>")
+			windoor.dir = dir
+			windoor.fingerprints += fingerprints
```
Comment (MadmanMartian) [inline code/game/objects/structures/windoor_assembly.dm]: Isn't there an addfingerprints or transferfingerprints function that can be used here?

Comment (jknpj) [reply code/game/objects/structures/windoor_assembly.dm]: That's only for mobs, we're copying the fingerprints from one object to another

File: code/game/machinery/doors/windowdoor.dm (hunk 1 above)
Comment (jknpj) [reply code/game/machinery/doors/windowdoor.dm]: Is it really necessary? Seemed to be fine when i was testing it(but i guess i forgot about hardmapped windoors)

File: code/game/objects/structures/windoor_assembly.dm (hunk 3 above)
Comment (I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE) [reply code/game/objects/structures/windoor_assembly.dm]: https://github.com/vgstation-coders/vgstation13/blob/Bleeding-Edge/code/game/atoms.dm#L739

Comment (jknpj) [reply code/game/objects/structures/windoor_assembly.dm]: Now THAT's the thing

**Comment by SonixApache** [comment]
any-order assembly is bad imo

**Pull request merged**